source/MapDetailPanel.cpp

```diff
@@ -573,13 +573,32 @@ void MapDetailPanel::DrawKey()
 		// four largest visible governments are labeled in the legend.
 		vector<pair<double, const Government *>> distances;
 		for(const auto &it : closeGovernments)
+		{
+			if(!it.first)
+				continue;
 			distances.emplace_back(it.second, it.first);
+		}
 		sort(distances.begin(), distances.end());
-		for(unsigned i = 0; i < 4 && i < distances.size(); ++i)
+		int drawn = 0;
+		vector<pair<const string, const Color *>> alreadyDisplayed;
```

Comment:
```suggestion
		vector<pair<string, const Color *>> alreadyDisplayed;
```

the const here inhibits move semantics